src/server/scripts/EasternKingdoms/ZulGurub/instance_zulgurub.cpp

```diff
@@ -122,16 +133,73 @@ class instance_zulgurub : public InstanceMapScript
                     case DATA_JINDO:
                         return _jindoTheHexxerGUID;
                         break;
-                    case NPC_ARLOKK:
+                    case DATA_ARLOKK:
                         return _arlokkGUID;
                         break;
+                    case DATA_HAKKAR:
+                        return _hakkarGUID;
+                        break;
                     case GO_GONG_OF_BETHEKK:
                         return _goGongOfBethekkGUID;
                         break;
                 }
                 return ObjectGuid::Empty;
             }
 
+            uint32 GetData(uint32 data) const override
+            {
+                switch (data)
+                {
+                    case DATA_HAKKAR_INTRO_1:
+                        return _hakkarIntro1;
+                        break;
+                    case DATA_HAKKAR_INTRO_2:
+                        return _hakkarIntro2;
+                        break;
+                    case DATA_HAKKAR_INTRO_3:
+                        return _hakkarIntro3;
+                        break;
+                }
+                return NOT_STARTED;
+            }
+
+            void SetData(uint32 data, uint32 value) override
+            {
+                switch (data)
+                {
+                    case DATA_HAKKAR_INTRO_1:
+                        _hakkarIntro1 = value;
+                        break;
+                    case DATA_HAKKAR_INTRO_2:
+                        _hakkarIntro2 = value;
+                        break;
+                    case DATA_HAKKAR_INTRO_3:
+                        _hakkarIntro3 = value;
+                        break;
+                }
+                if (value == DONE)
+                    SaveToDB();
+            }
+
+            void WriteSaveDataMore(std::ostringstream& oss) override
+            {
+                oss << "S " << uint32(_hakkarIntro1) << ' ' << uint32(_hakkarIntro2) << ' ' << uint32(_hakkarIntro3);
```

Comment:
i dont get why this 'S' is needed

Reply:
Copied that from Zul'Aman script, I think it makes sense to make sure the data is what you expect it to be.

Reply:
it realy not make any difference, you're justing saving a useless information xD